docker/docker.go

```diff
@@ -738,12 +759,22 @@ func isContainerEnvChanged(containerEnv, imageConfigEnv, dockerfileEnv []string)
 
 func GetKubeletDockerConfig(prsMap map[string]v3.PrivateRegistry) (string, error) {
 	auths := map[string]authConfig{}
-
+	credHelper := make(map[string]string)
 	for url, pr := range prsMap {
-		auth := base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%s:%s", pr.User, pr.Password)))
-		auths[url] = authConfig{Auth: auth}
+		if len(pr.CredentialPlugin) != 0 {
+			if credPluginType, ok := pr.CredentialPlugin["type"]; ok {
+				if credPluginType == "ecr" {
+					credHelper[pr.URL] = "ecr-login"
+				} else {
+					logrus.Infof("Skipping plugin in kubelet config.json. Plugin %s unsupported", credPluginType)
```

Comment:
Can we move this to `cluster/validation.go` and error out as its a non supported config.

Reply:
@superseb i have re-organized the code